utils/files.ts

```diff
@@ -0,0 +1,46 @@
+import { RESPONSE_MESSAGES } from "../constants/index";
+import { NextApiResponse } from "next";
+import { Transform } from "stream";
+
+export function resolveHeaders(headers:string[]): string {
+    return headers.join(',');
+}
+
+export function getDataFromValues(data:any, headers: string[]): any {
+    const result: any = {};
+    headers.forEach(header => {
+        result[header] = data[header];
+    });
+
+    return result;
+}
+
+export const getTransform = (headers:string[]) => new Transform({
+    objectMode: true,
+    transform(chunk: any, encoding: BufferEncoding, callback: () => void) {
+        const csvLine = headers.map(header => chunk[header]).join(',') + '\n';
+        this.push(csvLine);
+        callback();
+    },
+});
+
+export function streamtoCSV(values: object[], headers: string[], res: NextApiResponse | any) {
+    const maxRecords = 10; 
+    const formattedHeaders = resolveHeaders(headers);
```

Comment:
The variable is called `formattedHeaders`, the function `resolveHeaders` and what it does is join the headers with a comma.
So a few things here:
1. Consistent naming: the variable could be called `concatenatedHeaders` and the function `concatenateHeaders`, for example. Best even something like `concatenateColumns` since I guess we'd be using it to both headers and other rows.
2. Maybe use another delimiter than `,`?I suppose we'll have the Button Name in one of the columns, so best use a char that wouldn't appear there such as `\t`.